lua/octo/init.lua

```diff
@@ -761,4 +762,66 @@ function M.apply_buffer_mappings(bufnr, kind)
   end
 end
 
+function M.show_summary()
+  if vim.bo.ft ~= "octo_issue" then
+    return
+  end
+
+  local _, repo2 = pcall(api.nvim_buf_get_var, 0, "repo")
+  if not repo2 then
+    return
+  end
+
+  local cursor = api.nvim_win_get_cursor(0)
+  local current_line = vim.fn.getline(".")
+  local different_repo_issue_pattern = "%s([^ /]+/[^ #]+)#(%d+)%s"
+  local same_repo_issue_pattern = "%s#(%d+)%s"
+  local start_col, end_col, repo, number = current_line:find(different_repo_issue_pattern)
```

Comment:
This looks like some logic that is worth to get exported into its own function with a proper name. This reduces the number of abstraction levels of this function and improves the readability. Also this could/should become re-usable (e.g. for hover user names?).